sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/BlobItem.java

```diff
@@ -124,82 +154,91 @@ public BlobItem setSnapshot(String snapshot) {
     }
 
     /**
-     * Get the versionId property: The versionId property.
+     * Get the properties property: The properties property.
      *
-     * @return the versionId value.
+     * @return the properties value.
      */
-    public String getVersionId() {
-        return this.versionId;
+    public BlobItemProperties getProperties() {
+        return this.properties;
     }
 
     /**
-     * Set the versionId property: The versionId property.
+     * Set the properties property: The properties property.
      *
-     * @param versionId the versionId value to set.
+     * @param properties the properties value to set.
      * @return the BlobItem object itself.
      */
-    public BlobItem setVersionId(String versionId) {
-        this.versionId = versionId;
+    public BlobItem setProperties(BlobItemProperties properties) {
+        this.properties = properties;
         return this;
     }
 
     /**
-     * Get the isCurrentVersion property: The isCurrentVersion property.
+     * Get the metadata property: The metadata property.
      *
-     * @return the isCurrentVersion value.
+     * @return the metadata value.
      */
-    public Boolean isCurrentVersion() {
-        return this.isCurrentVersion;
+    public Map<String, String> getMetadata() {
+        return this.metadata;
     }
 
     /**
-     * Set the isCurrentVersion property: The isCurrentVersion property.
+     * Set the metadata property: The metadata property.
      *
-     * @param isCurrentVersion the isCurrentVersion value to set.
+     * @param metadata the metadata value to set.
      * @return the BlobItem object itself.
      */
-    public BlobItem setIsCurrentVersion(Boolean isCurrentVersion) {
-        this.isCurrentVersion = isCurrentVersion;
+    public BlobItem setMetadata(Map<String, String> metadata) {
+        this.metadata = metadata;
         return this;
     }
 
     /**
-     * Get the properties property: The properties property.
+     * Get the tags property: The tag property.
      *
-     * @return the properties value.
+     * @return the metadata value.
      */
-    public BlobItemProperties getProperties() {
-        return this.properties;
+    public Map<String, String> getTags() {
+        return this.tags;
     }
 
     /**
-     * Set the properties property: The properties property.
+     * Get the versionId property: The versionId property.
      *
-     * @param properties the properties value to set.
+     * @return the versionId value.
+     */
+    public String getVersionId() {
+        return this.versionId;
+    }
+
+    /**
+     * Set the versionId property: The versionId property.
+     *
+     * @param versionId the versionId value to set.
      * @return the BlobItem object itself.
      */
-    public BlobItem setProperties(BlobItemProperties properties) {
-        this.properties = properties;
+    public BlobItem setVersionId(String versionId) {
```

Comment:
Do we want to consider deprecating the setters if we no longer want people to use them?